Java/src/org/imos/ddb/JDBCDDB.java

```diff
@@ -33,160 +33,193 @@
+	{
+		this.driver = driver;
+		this.connection = connection;
+		this.user = user;
+		this.password = password;
+
+		// Test connection - throws exception at creation if can't connect
+		Class.forName(driver);
+
+		Connection conn = DriverManager.getConnection(connection, user, password);
+		
+		conn.close();
+	}
+
+	/**
+	 * Executes the given query, in the form:
+	 * 
+	 * select * from [tableName] where [fieldName] = '[fieldValue]'
+	 * 
+	 * The rows are converted into object equivalents of the table, and returned
+	 * in a List. If fieldName is null, the where clause is omitted, thus the
+	 * entire table is returned.
+	 * 
+	 * @param tableName
+	 *            The table to read.
+	 * 
+	 * @param fieldName
+	 *            the name of the query field.
+	 * 
+	 * @param fieldValue
+	 *            the query field value.
+	 * 
+	 * @throws Exception
+	 *             on any error.
+	 */
+	public List<Object> executeQuery(String tableName, String fieldName, Object fieldValue) throws Exception
+	{
+
+		Connection conn = null;
+		List<Object> results = null;
+
+		try
+		{
+
+			// create ODBC database connection
+			Class.forName(driver);
+
+			conn = DriverManager.getConnection(connection, user, password);
+
+			results = new ArrayList<Object>();
+
+			// build the query
+			String query = "SELECT * FROM " + tableName;
+			if (fieldName != null)
+			{
+
+				if (fieldValue == null)
+					throw new Exception("a fieldValue must be provided");
+
+				// wrap strings in quotes
+				if (fieldValue instanceof String)
+					query += " WHERE \"" + fieldName + "\" = '" + fieldValue + "'";
+				else
+					query += " WHERE \"" + fieldName + "\" = " + fieldValue;
+			}
+
+			// execute the query
+			Statement stmt = null;
+			ResultSet rs = null;
+			try
+			{
+				System.out.println("JDBCDDB::Query : " + query);
+				
+				stmt = conn.createStatement();
+				rs = stmt.executeQuery(query);
+			}
+			catch (Exception e)
+			{
+				System.out.println("JDBCDDB::Exception Caught " + e);
+				
+				// Hack to accommodate DeploymentId and FieldTripID
+				// types of number or text. Don't tell anyone
+				if (fieldName != null && fieldValue instanceof String)
+				{
+					query = "SELECT * FROM " + tableName + " WHERE \"" + fieldName + "\" = " + fieldValue;
+
+					System.out.println("JDBCDDB::Query : " + query);
+					
+					stmt = conn.createStatement();
+					rs = stmt.executeQuery(query);
+				}
+				else
+					throw e;
+			}
+
+			ResultSetMetaData rsmd = rs.getMetaData();
+			int columnsNumber = rsmd.getColumnCount();
+			int rows = 0;
+			// create an object for each row
+			while (rs.next())
+			{
+				rows++;
+				
+				ArrayList<Object> instance = new ArrayList<Object>();
+
+				results.add(instance);
+
+				for (int i = 1; i < columnsNumber; i++)
+				{
+					DBObject db = new DBObject();
+
+					db.name = rsmd.getColumnName(i);
+					db.o = rs.getObject(i);
+

```

Comment:
Create a DBObject with the name (from the column name) and the object from the database